components/workbenches/workbenches.go

```diff
@@ -27,6 +30,46 @@ type Workbenches struct {
 	components.Component `json:""`
 }
 
+func (w *Workbenches) DownloadManifests(platform string) error {
+	//Download manifests if defined by devflags
+	if len(w.DevFlags.Manifests) != 0 {
+		// Go through each manifests and set the overlays if defined
+		for _, subcomponent := range w.DevFlags.Manifests {
+			if strings.Contains(subcomponent.URI, DependentComponentName) {
```

Comment:
2 parts need to consider:
- after the transition, workbench will only use "kubeflow" git repo,
- user who forked the kubeflow can have a different repo name, e.g
```
workbenches:
      devFlags:
        manifests:
          - manifestsFolder: component/notebook-controller
            overlay: base
            uri: 'https://github.com/my-repo/my-kubeflow/tarball/some-branch'
          - manifestsFolder: component/odh-notebook-controller
            overlay: overlays/openshift
            uri: >-
              https://github.com/your-repo/your-kubeflow/tarball/another-branch
```

any reason we need to  have this check on `strings.Contains(subcomponent.URI,...)`

Reply:
in my PR to get nbc working after trantiion,
i renamed it from "notebook-controller" to the folder "kf-notebook-controller" in the operator image.
e.g 
`    "kubeflow:v1.7-branch:components/notebook-controller/config:odh-notebook-controller/kf-notebook-controller"`

so for workbench, the case might be to check s`trings.Contains(manifestsFolder, "odh-notebook-controller")`
and `strings.Contains(manifestsFolder, "kf-notebook-controller")`

Reply:
This implementation doesn't support multiple versions of same manifests, we have tight coupling with deployment paths.

The manifests list just state if there are multiple sub-components, example `kubeflow` and `notebooks`, a user can list them.